Utils/Dataflow/009_oracleConnector/Oracle2JSON.py

```diff
@@ -476,9 +483,10 @@ def process(conn, offset_storage, config):
     offset_date = get_offset(offset_storage)
     if not reverse and not final_date:
         # In case of normal data load 'final_date' may be None:
-        # it means we need to adjust it to current timestamp
-        # before every check
-        final_date = offset_now()
+        # it means we need to set it to current timestamp
+        # before every check. 'delay' can be used to adjust the resulting
+        # final_date in this case.
+        final_date = offset_now() - timedelta(seconds=config['delay'])
```

Comment:
Seems like same operation should be done everywhere `offset_now()` is used.
E.g. here, as "right" border of time interval is adjusted within the loop:
https://github.com/PanDAWMS/dkb/blob/715d86ff923058ea5ae9605f823b806988d709ac/Utils/Dataflow/009_oracleConnector/Oracle2JSON.py#L518-L519

Maybe it should be done in within the function, just like time zone is taken into account only there? It would require global variable, of course (and there\`s nothing good in it), yet will allow to "automatically" take the delay into account whenever we need "current timestamp".

Reply:
I agree with

> Seems like same operation should be done everywhere offset_now() is used.
> ...
> Maybe it should be done in within the function, just like time zone is taken into account only there?

, but do we really need global variable? Delay is used for a single purpose, adjusting the "right" boundary of the time interval. At the moment it is done 3 times (via `offset_now()`), once in `read_config()` and twice in `process()`, both functions have access to config and delay. Maybe making `offset_now()` accept delay as a parameter will be enough? Or it is highly possible in the future to encounter a situation where `offset_now()` must be called, but config is unavailable?

Reply:
> Maybe making `offset_now()` accept delay as a parameter will be enough? Or it is highly possible in the future to encounter a situation where `offset_now()` must be called, but config is unavailable?

I\`d say that it is much more possible that even when `config` is available, one can simply forget to pass `delay` (if, of course, `offset_now()` does not require both parameters to be excplicitly passed.

My point is that `delay` must affect the behaviour of the whole stage: whenever it uses current timestamp, it must be adjusted with `delay`.

Actually, the better way would be to wrap all the stage this to some kind of object, where `config` would be nothing but an instance variable. It is way too much, of course ;) But still made me think that maybe instead of global `OFFSET_TZ` and possible `DELAY` we might want to switch to global `CONFIG` (it might, by the way, had already been global once, but turned into local as superfluous).

Reply:
Fair enough. Made adjustments to the first commit and force-pushed the branch. Please, take a look.